### PR TITLE
Enable openshift e2e tests on centos

### PR DESCRIPTION
**Pull request by kargakis**
@jim-minter @jackfrancis @pweil-

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -529,6 +566,10 @@ workflows:
           filters:
             branches:
               only: master
+      - openshift-3.9-centos-e2e:
```
Comment (CecileRobertMichon) [inline .circleci/config.yml]: Missing
           requires:
             - test

Comment (kargakis) [reply .circleci/config.yml]: fixed

**Comment by CecileRobertMichon** [comment]
@kargakis the openshift centos e2e test failed with:
``` 
$ az group deployment create --name openshift-eastus-14694 --resource-group openshift-eastus-14694 --template-file /go/src/github.com/Azure/acs-engine/_output/openshift-eastus-14694/azuredeploy.json --parameters /go/src/github.com/Azure/acs-engine/_output/openshift-eastus-14694/azuredeploy.parameters.json
2018/05/11 18:29:04 
Error from deployment for openshift-eastus-14694 in resource group openshift-eastus-14694:exit status 1
2018/05/11 18:29:04 Command Output: ERROR: Marketplace purchase eligibilty check returned errors. See inner errors for details
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -384,6 +384,37 @@ jobs:
...
+        environment:
+          GOPATH: /go
+    steps:
+      - checkout
+      - run: |
+          echo 'export TIMEOUT=30m' >> $BASH_ENV
+          echo 'export DISTRO=openshift39_centos' >> $BASH_ENV
+          echo 'export LOCATION=eastus' >> $BASH_ENV
+          echo 'export ORCHESTRATOR_RELEASE=3.9' >> $BASH_ENV
+          echo 'export CLUSTER_DEFINITION=examples/openshift.json' >> $BASH_ENV
```
Comment (kargakis) [inline .circleci/config.yml]: @jim-minter this is how you configure the api model

**Comment by kargakis** [comment]
I'll look into the failure on Monday but it looks like an issue with the CI subscription not being able to access the centos image.

**Comment by kargakis** [comment]
@kwoodson can you walk me through how to check the enabled subscriptions in the centos image?

**Comment by kargakis** [comment]
Also, seeing a permission issue when running locally that needs to be fixed before we merge this

https://github.com/Azure/acs-engine/issues/2960

**Comment by kargakis** [comment]
@CecileRobertMichon can you please restart the test?

**Comment by CecileRobertMichon** [comment]
@kargakis it failed again with ```
$ az group deployment create --name openshift-eastus-40186 --resource-group openshift-eastus-40186 --template-file /go/src/github.com/Azure/acs-engine/_output/openshift-eastus-40186/azuredeploy.json --parameters /go/src/github.com/Azure/acs-engine/_output/openshift-eastus-40186/azuredeploy.parameters.json
2018/05/16 16:18:55 
Error from deployment for openshift-eastus-40186 in resource group openshift-eastus-40186:exit status 1
2018/05/16 16:18:55 Command Output: ERROR: Marketplace purchase eligibilty check returned errors. See inner errors for details. ```

**Comment by kargakis** [comment]
@kwoodson any idea why we are seeing marketplace errors when using the published centos image?

**Comment by kargakis** [comment]
Here are the image fields used for the centos image:
https://github.com/Azure/acs-engine/blob/bcda440129a9c662fc44499e5d37d2acd960dda9/pkg/acsengine/defaults.go#L96-L101

**Comment by kargakis** [comment]
@CecileRobertMichon ping

**Comment by CecileRobertMichon** [comment]
@kargakis the latest test failed with a vm extension provisioning error so I restarted it

**Comment by CecileRobertMichon** [comment]
@kargakis please rebase

/lgtm

**Comment by kargakis** [comment]
@CecileRobertMichon rebased, ptal

/unassign

**Pull request merged**